File: Python/Calculator/CalculatorServer/Services/CalculatorService.py
import calculator_pb2
import calculator_pb2_grpc
from Utils.Calculator import Calculator

class CalculatorService(calculator_pb2_grpc.CalculatorServiceServicer):
    
    def Calculate(self, request_iterator, context):
        calculator = Calculator()
        for request in request_iterator:
            if request.operand is not None:
                calculator.add_operand(request.operand.value)
            if calculator.can_calculate():
                if request.operation != calculator_pb2.InvalidOperation:
                    print(f'The following operation will be calculated: {request.operation}')
                    yield calculator_pb2.OperationResponse(result = calculator.calculate(request.operation), message = f'Calculated operation: {request.operation}.')
                else:
                    yield calculator_pb2.OperationResponse(result = float(), message = 'Please enter an operation type.')
            else:
                yield calculator_pb2.OperationResponse(result = float(), message = 'Please enter one more operand.')

File: .NET Core 3.0/Calculator/CalculatorServer/Services/CalculatorService.cs
using CalculatorServer.Actions;
using Generated;
using Grpc.Core;
using NLog;
using System;
using System.Threading.Tasks;

namespace CalculatorServer.Services
{
  internal sealed class CalculatorService : Generated.CalculatorService.CalculatorServiceBase
  {
    #region Logger
    private static Logger Logger { get; } = LogManager.GetCurrentClassLogger();
    #endregion
    public override async Task Calculate(IAsyncStreamReader<OperationRequest> requestStream, IServerStreamWriter<OperationResponse> responseStream, ServerCallContext context)
    {
      Calculator calculator = new Calculator();
      while (await requestStream.MoveNext())
      {
        var request = requestStream.Current;
        if (request.Operand != null)
        {
          calculator.AddOperand(request.Operand.Value);
        }
        if (calculator.CanCalculate)
        {
          if (request.Operation != OperationType.InvalidOperation)
          {
            Console.WriteLine($"The following operation will be calculated: {request.Operation}");
            await SendResultResponse(responseStream, calculator, request);
          }
          else
          {
            await SendOperationNeededResponse(responseStream);
          }
        }
        else
        {
          await SendOperandNeededResponse(responseStream);
        }
      }
    }

    #region Private methods
    private static async Task SendResultResponse(IServerStreamWriter<OperationResponse> responseStream, Calculator calculator, OperationRequest request)
    {
      await responseStream.WriteAsync(new OperationResponse()
      {
        Result = calculator.Calculate(request.Operation),
        Message = $"Calculated operation: {request.Operation}."
      });
    }

    private static async Task SendOperationNeededResponse(IServerStreamWriter<OperationResponse> responseStream)
    {
      await responseStream.WriteAsync(new OperationResponse()
      {
        Result = default,
        Message = "Please enter an operation type."
      });
    }

    private static async Task SendOperandNeededResponse(IServerStreamWriter<OperationResponse> responseStream)
    {
      await responseStream.WriteAsync(new OperationResponse()
      {
        Result = default,
        Message = "Please enter one more operand."
      });
    }
    #endregion
  }
}
